src/Microsoft.AspNet.Mvc.Razor/DefaultViewLocationCache.cs

```diff
@@ -50,9 +50,11 @@ internal static string GenerateKey(ViewLocationExpanderContext context)
             var routeValues = context.ActionContext.RouteData.Values;
             var controller = routeValues.GetValueOrDefault<string>(RazorViewEngine.ControllerKey);
 
-            // format is "{viewName}:{controllerName}:{areaName}:"
+            // format is "{viewName}:{isPartial}:{controllerName}:{areaName}:"
             keyBuilder.Append(context.ViewName)
                       .Append(CacheKeySeparator)
+                      .Append(context.IsPartial ? 1 : 0)
```

Comment:
Consider if it's worth using something other than a digit here to keep this readable. I don't think it matters much but the thought occurred to me.